File: debug_tools/test_ollama.py
import requests
import json

def test_ollama():
    url = "http://localhost:11434/api/generate"
    
    # Test data
    payload = {
        "model": "llama2",
        "prompt": "correct this for grammar and typos, maintain the same tone in the resulting text:this is a test sentense with some typos and grammer mistakes",
        "stream": False
    }
    
    print("Sending request with payload:")
    print(json.dumps(payload, indent=2))
    
    try:
        response = requests.post(url, json=payload)
        print(f"\nResponse Status Code: {response.status_code}")
        print("Response Headers:")
        print(json.dumps(dict(response.headers), indent=2))
        
        if response.status_code == 200:
            print("\nResponse Content:")
            print(response.text)
            
            # Try to parse the response
            try:
                json_response = response.json()
                print("\nParsed JSON Response:")
                print(json.dumps(json_response, indent=2))
                print("+"*20)
                print("So in conclusion:")
                print(json_response["response"])
            except json.JSONDecodeError as e:
                print(f"\nCould not parse response as JSON: {e}")
                print("Raw response content:")
                print(response.text)
        else:
            print(f"Error: Received status code {response.status_code}")
            print(response.text)
            
    except requests.exceptions.ConnectionError:
        print("Connection Error: Could not connect to Ollama server. Is it running?")
    except Exception as e:
        print(f"Unexpected error: {e}")

if __name__ == "__main__":
    print("Testing Ollama API...")
    test_ollama()

File: debug_tools/test_ollama.js
const fetch = require('node-fetch');

async function testOllama() {
    const url = "http://localhost:11434/api/generate";
    
    // Test data matching your background.js
    const payload = {
        model: "llama2",
        prompt: "correct this for grammar and typos, maintain the same tone in the resulting text this is a test sentence with typos",
        stream: false
    };
    
    console.log("Sending request with payload:");
    console.log(JSON.stringify(payload, null, 2));
    
    try {
        const response = await fetch(url, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(payload)
        });

        console.log(`\nResponse Status Code: ${response.status}`);
        console.log("Response Headers:");
        console.log(JSON.stringify(Object.fromEntries([...response.headers]), null, 2));
        
        if (response.status === 200) {
            const text = await response.text();
            console.log("\nResponse Content:");
            console.log(text);
            
            // Try to parse the response
            try {
                const jsonResponse = JSON.parse(text);
                console.log("\nParsed JSON Response:");
                console.log(JSON.stringify(jsonResponse, null, 2));
            } catch (e) {
                console.log("\nCould not parse response as JSON:", e.message);
                console.log("Raw response content:");
                console.log(text);
            }
        } else {
            console.log(`Error: Received status code ${response.status}`);
            const text = await response.text();
            console.log(text);
        }
    } catch (error) {
        if (error.code === 'ECONNREFUSED') {
            console.log("Connection Error: Could not connect to Ollama server. Is it running?");
        } else {
            console.log("Unexpected error:", error);
        }
    }
}

console.log("Testing Ollama API...");
testOllama();
